src/main/scala/scalafix/sbt/ScalafixEnable.scala

```diff
@@ -2,66 +2,184 @@ package scalafix.sbt
 import sbt.Keys._
+import sbt.VersionNumber.SemVer
+
+import ScalafixPlugin.autoImport.scalafixResolvers
+
+import collection.JavaConverters._
+import scala.util._
+import coursierapi.Repository
 
 /** Command to automatically enable semanticdb compiler output for shell session
   */
 object ScalafixEnable {
 
-  /** If the provided Scala binary version is supported, return the full version
-    * required for running semanticdb-scalac.
+  /** If the provided Scala binary version is supported, return the latest scala
+    * full version for which the recommended semanticdb-scalac is available
     */
-  private lazy val semanticdbScalacFullScalaVersion
-      : PartialFunction[(Long, Long), String] = (for {
+  private lazy val recommendedSemanticdbScalacScalaVersion
+      : PartialFunction[(Long, Long), VersionNumber] = (for {
     v <- BuildInfo.supportedScalaVersions
     p <- CrossVersion.partialVersion(v).toList
-  } yield p -> v).toMap
+  } yield p -> VersionNumber(v)).toMap
 
-  /** If the provided Scala binary version is supported, return the full version
-    * required for running semanticdb-scalac or None if support is built-in in
-    * the compiler and the full version does not need to be adjusted.
+  /** If the provided Scala binary version is supported, return the latest scala
+    * full version for which the recommended semanticdb-scalac is available, or
+    * None if semanticdb support is built-in in the compiler
     */
-  private lazy val maybeSemanticdbScalacFullScalaVersion
-      : PartialFunction[(Long, Long), Option[String]] =
-    semanticdbScalacFullScalaVersion.andThen(Some.apply).orElse {
+  private lazy val maybeRecommendedSemanticdbScalacScalaVersion
+      : PartialFunction[(Long, Long), Option[VersionNumber]] =
+    recommendedSemanticdbScalacScalaVersion.andThen(Some.apply).orElse {
       // semanticdb is built-in in the Scala 3 compiler
       case (major, _) if major == 3 => None
     }
 
-  /** Collect projects across the entire build, using the partial function
-    * accepting a Scala binary version
-    */
-  private def collectProjects[U](
-      extracted: Extracted,
-      pf: PartialFunction[(Long, Long), U]
-  ): Seq[(ProjectRef, U)] = for {
-    p <- extracted.structure.allProjectRefs
-    version <- (p / scalaVersion).get(extracted.structure.data).toList
-    partialVersion <- CrossVersion.partialVersion(version).toList
-    res <- pf.lift(partialVersion).toList
-  } yield p -> res
+  /** Collect compatible projects across the entire build */
+  private def collectProjects(extracted: Extracted): Seq[CompatibleProject] =
+    for {
+      p <- extracted.structure.allProjectRefs
+      scalaV <- (p / scalaVersion).get(extracted.structure.data).toList
+      partialVersion <- CrossVersion.partialVersion(scalaV).toList
+      maybeRecommendedSemanticdbScalacV <-
+        maybeRecommendedSemanticdbScalacScalaVersion.lift(partialVersion).toList
+      scalafixResolvers0 <- (p / scalafixResolvers)
```

Comment:
this is normally set at the `ThisBuild` level, but since usage of custom resolvers is not really tested, I chose to be overly defensive